File: 백준/Gold/1263. 시간 관리/시간 관리.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Comparator;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        int[][] work = new int[n][2];

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            work[i][0] = Integer.parseInt(st.nextToken());
            work[i][1] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(work, (a, b) -> b[1] - a[1]);

        int start = work[0][1];

        for (int[] arr : work) {
            start = Math.min(start, arr[1]) - arr[0];
        }

        System.out.println(Math.max(-1, start));
    }
}

File: 백준/Gold/1263. 시간 관리/시간 관리.py
n = int(input())
work = []

for _ in range(n):
    t, s = map(int, input().split())
    work.append((t, s))

work.sort(key=lambda x: -x[1])

start = work[0][1]

for t, s in work:
    start = min(start, s) - t

print(max(-1, start))